app/assets/javascripts/services/miq_service.js

```diff
@@ -175,4 +175,32 @@ ManageIQ.angular.app.service('miqService', ['$timeout', '$document', '$q', 'API'
 
     $window.location.href = redirectUrl;
   };
+
+  this.autoUpdateDashboard = function($window, $scope, $http, $interval, base_url, callback) {
+    $scope.refresh = function() {
+      // get the pathname and remove trailing / if exist
+      var pathname = $window.location.pathname.replace(/\/$/, '');
+      if (pathname.match(/show$/)) {
+        $scope.id = '';
+      } else if (pathname.match(/^\/[^\/]+\/show\/(\d+)/)) {
+        // search for pattern ^/<controler>/show/<id>$ in the pathname
+        $scope.id = '/' + (/^\/[^\/]+\/show\/(\d+)/.exec(pathname)[1]);
+      } else {
+        // search for pattern ^/<controler>/<id>$ in the pathname
+        $scope.id = '/' + (/^\/[^\/]+\/(\d+)$/.exec(pathname)[1]);
+      }
+
+      var url = base_url + $scope.id;
+      $http.get(url)
+        .then(callback)
+        .catch(miqService.handleFailure);
+    };
+
+    $scope.refresh();
+    var promise = $interval($scope.refresh, 1000 * 60 * 3);
+
+    $scope.$on('$destroy', function() {
+      $interval.cancel(promise);
+    });
+  }
```

Comment:
Im not too sure this belongs in `MiqService`

Reply:
agreed, we should create a DashboardService or something for these..